mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,383 @@
+        )
+        self.title = os.path.splitext(self._track_paths["annotations"][0])[0]
+
+    @core.cached_property
+    def score(self) -> music21.stream.Score:
+        return load_score(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys(self) -> Optional[KeyData]:
+        return load_key(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys_music21(self) -> Optional[List[dict]]:
+        return load_key_music21(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def roman_numerals(self) -> Optional[List[dict]]:
+        return load_roman_numerals(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def chords(self) -> Optional[ChordData]:
+        return load_chords(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def chords_music21(self) -> Optional[List[dict]]:
+        return load_chords_music21(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def duration(self) -> int:
+        return self.chords_music21[-1]["time"]
+
+    @core.cached_property
+    def midi_path(self) -> Optional[str]:
+        return load_midi_path(self.humdrum_annotated_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            metadata={
+                "duration": self.duration,
+                "title": self.title,
+                "key": self.keys,  # format is not the expected by keydata jams namespace
+                "chord": self.chords,  # format is not the expected by chorddata jams namespace
+                "keys_music21": self.keys_music21,
+                "chords_music21": self.chords_music21,
+                "roman_numerals": self.roman_numerals,
+                "midi_path": self.midi_path,
+                "humdrum_annotated_path": self.humdrum_annotated_path,
+            },
+        )
+
+
+def split_score_annotations(fhandle: TextIO):
+    """Load haydn op20 score and annotations divided.
+
+    Args:
+        fhandle (str or file-like): path to hrm annotations
+
+    Returns:
+        music21.stream.Score: score in music21 format
+        List[Tuple[Any, Any]]: roman numerals
+    """
+    score = music21.converter.parse(fhandle.name, format="humdrum")
+
+    rna = {rn.offset: rn for rn in list(score.flat.getElementsByClass("RomanNumeral"))}
+    score.remove(rna, recurse=True)
+    rna_clean = [(offset, rn) for offset, rn in rna.items() if rn]
+    return score, rna_clean
+
+
+@io.coerce_to_string_io
+def load_score(fhandle: TextIO):
+    """Load haydn op20 score with annotations from a file with music21 format (music21.stream.Score).
+
+    Args:
+        fhandle (str or file-like): path to score
+
+    Returns:
+        music21.stream.Score: score in music21 format
+
+    """
+    score, rna = split_score_annotations(fhandle)
+    return score
+
+
+def load_key_base(fhandle, resolution):
```

Comment:
Can you make this a hidden function `_load_key_base`

Reply:
done